tests/utils/command_test_utils.go

```diff
@@ -34,79 +47,81 @@ func addAttributes(numAtrributes int) map[string]string {
 }
 
 // addGroup creates and returns a group in a schema structure
-func addGroup() *schema.CommandGroup {
+func (devfile *TestDevfile) addGroup() *schema.CommandGroup {
 
 	commandGroup := schema.CommandGroup{}
 	commandGroup.Kind = GetRandomGroupKind()
-	LogInfoMessage(fmt.Sprintf("group Kind: %s", commandGroup.Kind))
-	commandGroup.IsDefault = GetBinaryDecision()
+	LogInfoMessage(fmt.Sprintf("group Kind: %s, default already set %t", commandGroup.Kind, devfile.GroupDefaults[commandGroup.Kind]))
+	// Ensure only one and at least one of each type are labelled as default
+	if !devfile.GroupDefaults[commandGroup.Kind] {
+		devfile.GroupDefaults[commandGroup.Kind] = true
+		commandGroup.IsDefault = true
+	} else {
+		commandGroup.IsDefault = false
+	}
 	LogInfoMessage(fmt.Sprintf("group isDefault: %t", commandGroup.IsDefault))
 	return &commandGroup
 }
 
-// AddCommand adds a command of the specified type, with random attributes, to the devfile schema
-func (devfile *TestDevfile) AddCommand(commandType schema.CommandType) string {
-	command := generateCommand(commandType)
-	devfile.SchemaDevFile.Commands = append(devfile.SchemaDevFile.Commands, command)
-	return command.Id
-}
-
-// generateCommand creates a command of a specified type in a schema structure
-func generateCommand(commandType schema.CommandType) schema.Command {
-	command := schema.Command{}
-	command.Id = GetRandomUniqueString(8, true)
-	LogInfoMessage(fmt.Sprintf("command Id: %s", command.Id))
+// AddCommand creates a command of a specified type in a schema structure and pupulates it with random attributes
+func (devfile *TestDevfile) AddCommand(commandType schema.CommandType) schema.Command {
 
+	var command *schema.Command
```

Comment:
perhaps create type of `schema.Command`, and pass in `&command`
can use a switch case here. depends on your preference

Reply:
I kind of like it how it is but can looks at it more. OK to defer to later?